test/Tests/src/Semantic/Conversion_Spec.enso

```diff
@@ -112,4 +112,7 @@ spec =
             meta_from.rename "foo" 123 . should_equal "foo called"
             meta_from.rename "foo" . should_equal .foo
 
+        Test.specify "should not allow currying" <|
+            Panic.recover Any (Foo.from) . catch Any .to_display_text . should_equal "Conversion currying without `that` argument is not supported."
```

Comment:
Wouldn't `should_fail_with` be better suited for the test? Or do you want the test to pass even when no exception happen?

Reply:
> Wouldn't should_fail_with be better suited for the test?

Actually that won't work with Panics.

Reply:
I would appreciate if we do test the message.

Reply:
`should_fail_with` will work if you first do `Panic.recover` - `recover` will convert the Panic into dataflow error and then `should_fail_with` can handle it.

Reply:
And using `catch` you can then check the error message. Like this:
```suggestion
            r = Panic.recover Any (Foo.from)
            r.should_fail_with No_Conversion_Currying
            r.catch . to_display_text . should_equal "Conversion currying without `that` argument is not supported."
```

Reply:
Alternatively, you can use `Test.expect_panic_with` directly - this is a function meant for testing panics. But it won't allow you to check the error message and I think checking it is good - we had lots of places where `to_display_text` would crash because of typos and that went undetected only until the IDE was unable to display error properly. We want to go in the direction of having `to_display_text` of every error unit tested to ensure no regressions of error display can get into the IDE.